src/engine/QueryExecutionTree.cpp

```diff
@@ -104,6 +104,13 @@ size_t QueryExecutionTree::getSizeEstimate() {
   return sizeEstimate_.value();
 }
 
+//_____________________________________________________________________________
+void QueryExecutionTree::setPrefilterExpression(
+    const std::vector<Operation::PrefilterVariablePair>& prefilterVec) const {
+  AD_CONTRACT_CHECK(rootOperation_);
+  rootOperation_->setPrefilterExprGetUpdatedQetPtr(prefilterVec);
```

Comment:
This function can filter the visible variables etc, and should be called by the Filter class.